From akpm@osdl.org Mon Nov 14 23:38:45 2005
Message-Id: <200511150751.jAF7p9fP022460@shell0.pdx.osdl.net>
To: akpm@osdl.org, greg@kroah.com, mm-commits@vger.kernel.org
From: akpm@osdl.org
Subject: usb devio warning fix
Date: Mon, 14 Nov 2005 23:50:53 -0800

From: Andrew Morton <akpm@osdl.org>

drivers/usb/core/devio.c: In function `proc_ioctl_compat':
drivers/usb/core/devio.c:1401: warning: passing arg 1 of `compat_ptr' makes integer from pointer without a cast

Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/usb/core/devio.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- gregkh-2.6.orig/drivers/usb/core/devio.c
+++ gregkh-2.6/drivers/usb/core/devio.c
@@ -1398,7 +1398,7 @@ static int proc_ioctl_compat(struct dev_
 	struct usbdevfs_ioctl ctrl;
 	u32 udata;
 
-	uioc = compat_ptr(arg);
+	uioc = compat_ptr((long)arg);
 	if (get_user(ctrl.ifno, &uioc->ifno) ||
 	    get_user(ctrl.ioctl_code, &uioc->ioctl_code) ||
 	    __get_user(udata, &uioc->data))
